service/controller/v2/resource/node/create.go

```diff
@@ -114,30 +123,58 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 
 	if len(customPods) > 0 {
 		r.logger.LogCtx(ctx, "level", "debug", "message", "deleting all pods running custom workloads")
+		t1 := time.Now()
 
+		var wg sync.WaitGroup
 		for _, p := range customPods {
-			err := k8sClient.CoreV1().Pods(p.GetNamespace()).Delete(p.GetName(), &apismetav1.DeleteOptions{})
-			if err != nil {
-				return microerror.Mask(err)
-			}
+			// delete each pod in separate goroutine to avoid blocking
+			wg.Add(1)
+			go func(k8sClient kubernetes.Interface, pod v1.Pod) {
```

Comment:
I would like to use https://godoc.org/golang.org/x/sync/errgroup for parallel execution. It is our standard way of doing it and has better error handling capabilities. Also note the variable scope binding in loops. This is very important. The code is broken right now already and only the last pod will be managed.